pyccel/parser/semantic.py

```diff
@@ -2198,6 +2199,8 @@ def _visit_Assign(self, expr, **settings):
                 results = func.results
                 if results:
                     if len(results)==1:
+                        if isinstance(results[0], Variable) and results[0].allocatable:
+                           results[0].allocatable = False
```

Comment:
What is an array that is not `allocatable`, nor a `stack_array`, nor `is_pointer`? If it is not `allocatable` then you can't use `allocate` to define its size